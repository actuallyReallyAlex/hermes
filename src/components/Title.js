import React from 'react'
import { Heading } from 'grommet'

const Title = () => {
  return <Heading level="1">Hermes</Heading>
}

export default Title
