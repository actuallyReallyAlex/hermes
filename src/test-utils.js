import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { defaultState } from './fixtures'
import { render } from '@testing-library/react'

const mockThunks = ({ dispatch, getState }) => next => action => {
  const mockedAction = () => ({ type: 'MOCKED ACTION' })

  let returnValue

  if (typeof action === 'function' || action === undefined) {
    returnValue = mockedAction
  } else {
    returnValue = next(action)
  }

  return returnValue
}

const middlewares = [mockThunks]

export const mockStore = configureStore(middlewares)

export const customRender = options => {
  const { component, state, props } = options

  const Component = component

  return render(
    <Provider store={mockStore(state ? state : defaultState)}>
      <Component {...props} />
    </Provider>
  )
}
