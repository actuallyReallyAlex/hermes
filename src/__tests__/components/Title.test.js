import Title from '../../components/Title'
import { customRender } from '../../test-utils'

describe('<Title />', () => {
  it('Should render the <Title /> component.', () => {
    const container = customRender({ component: Title })
    expect(container.asFragment()).toMatchSnapshot()
  })
})
