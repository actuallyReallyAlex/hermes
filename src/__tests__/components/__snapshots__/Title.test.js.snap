// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Should render the <Title /> component. 1`] = `"<h1 class=\\"StyledHeading-sc-1rdh4aw-0 rfVJR\\">Hermes</h1>"`;
